src/common/dao/webhook_policy.go

```diff
@@ -0,0 +1,61 @@
+package dao
+
+import (
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/models"
+)
+
+// GetWebhookPolicy return webhook policy by id
+func GetWebhookPolicy(id int64) (*models.WebhookPolicy, error) {
+	policy := new(models.WebhookPolicy)
+	o := orm.NewOrm()
+	err := o.QueryTable(policy).Filter("id", id).One(policy)
+	if err == orm.ErrNoRows {
+		return nil, nil
+	}
+	return policy, err
+}
+
+// GetWebhookPolicyByName return webhook policy by name
+func GetWebhookPolicyByName(name string, projectID int64) (*models.WebhookPolicy, error) {
+	policy := new(models.WebhookPolicy)
+	o := GetOrmer()
+	err := o.QueryTable(policy).Filter("name", name).Filter("projectID", projectID).One(policy)
+	if err == orm.ErrNoRows {
+		return nil, nil
+	}
+	return policy, err
+}
+
+// GetWebhookPolicies returns all webhook policy in project
+func GetWebhookPolicies(projectID int64) (int64, []*models.WebhookPolicy, error) {
```

Comment:
Do we need to return total?  Isn't the length of the slice `[]*models.WebhookPolicy` total?

Reply:
> Since you are not handling the pagination, do we need to return total? Isn't the length of the slice `[]*models.WebhookPolicy` total?

will fix it

Reply:
It's still not fixed...